packages/token-bridge-sdk/src/hooks/useArbTokenBridge.ts

```diff
@@ -683,6 +737,38 @@ export const useArbTokenBridge = (
     }
   }
 
+  async function verifyDestinationAddressOrThrow({
+    provider,
+    destinationAddress,
+    isDeposit
+  }: {
+    provider?: Provider
+    destinationAddress?: string
+    isDeposit: boolean
+  }) {
```

Comment:
We already have [this](https://github.com/OffchainLabs/arbitrum-token-bridge/blob/master/packages/arb-token-bridge-ui/src/util/AddressUtils.ts#L3) and we should use it. The function is a bit cluttered and it it's not really pure as it also reads a bunch of outer state so I'm thinking we could split it up a bit:

```
async function verifyDepositOrThrow({
  from,
  to,
}: {
  from: string
  to: string
})

async function verifyWithdrawalOrThrow({
  from,
  to,
}: {
  from: string
  to: string
})
```

That way we don't have to worry about which providers to use and have logic to check for that inside the function.